utils/python/CIME/XML/testlist.py

```diff
@@ -64,16 +64,16 @@ def _get_testsv2(self, machine=None, category=None, compiler=None):
 
         for tnode in testnodes:
             machnodes = self.get_nodes("machine",machatts,root=tnode)
-            thistest = {}
-
             if machnodes:
+                thistnode = {}
                 for key, value in tnode.items():
                     if key == "name":
-                        thistest["testname"] = value
+                        thistnode["testname"] = value
```

Comment:
Should be thisnode instead of thistnode? Also, underscores are nice to seperate words in varnames. this_is_more_readable than thisislessreadable.